sql/src/main/java/io/crate/executor/transport/executionphases/SetBucketAction.java

```diff
@@ -33,14 +33,14 @@
     private final List<PageBucketReceiver> pageBucketReceivers;
     private final int bucketIdx;
     private final InitializationTracker initializationTracker;
-    private final BucketResultListener bucketResultListener;
+    private final PagingUnsupportedResultListener pagingUnsupportedResultListener;
 
     SetBucketAction(List<PageBucketReceiver> pageBucketReceivers, int bucketIdx, InitializationTracker initializationTracker) {
         assert !pageBucketReceivers.isEmpty() : "pageBucketReceivers must not be empty";
         this.pageBucketReceivers = pageBucketReceivers;
         this.bucketIdx = bucketIdx;
         this.initializationTracker = initializationTracker;
-        bucketResultListener = new BucketResultListener();
+        pagingUnsupportedResultListener = PagingUnsupportedResultListener.INSTANCE;
```

Comment:
do we need to keep it in a local variable, Since now it's a "singleton" ?

Reply:
No, I'm changing that in https://github.com/crate/crate/pull/6767 - but can already change it within this PR if you want me to

Reply:
No it's ok, thanks!